### ISSUE TITLE
Testing against rachel's examples

### ISSUE THREAD

**Issue by delebash** [FremyCompany/css-grid-polyfill#13]
Env: Chrome 42.02311.11 dev-m x64 without experimental features enable, Windows 7 x64
First nice job.  Just wanted to give some feedback
http://gridbyexample.com/examples/
Examples that do not match rachel's,  
#1, not sure what's going on here
#11 and #12 the . for empty track is not working, but it works in other examples, maybe because these are both grid areas
#19 the outline is not showing correctly
#20, this is new so not expected to work auto keyword

**Comment by FremyCompany** [comment]
Thanks for taking time to make this report. Unfortunately for you, I think most of those differences are actually issues in Chrome's implementation and not mine ;-)

---------------------

For instance, Ex1 was deemed wrong in Chrome by the spec editor here: http://lists.w3.org/Archives/Public/www-style/2014Nov/0003.html (the background around C should be drawn over B and therefore hide it to the user). 

---------------------

I don't think Ex11 and Ex12 are really bugs on my part, it's basically because the element is too big for the column Rachel defined. In Chrome, it shrinks the element so much that the text ends up being drawn over the element's padding. My reading of the spec is that you should not shrink-to-fit an element so the bug is in Chrome here (it didn't follow a spec change). See http://lists.w3.org/Archives/Public/www-style/2014Nov/0007.html for details.

---------------------

The "outline bug" in Ex19 is not actually a bug, because where browsers draw the outline is NOT actually defined in a spec (in fact, it has been decided by the csswg this would be left undefined on purpose as recently as last week see CSS-UI in https://lists.w3.org/Archives/Public/www-style/2015Feb/0504.html). 

I agree the current rendering is unfortunate, though, but there's not much I can do to fix it given the outline is drawn by the browser. I'll discuss with the developers behind css-grid in Chrome how they do the drawing, I think they are using an inline drawing algorithm and that's probably going to change once they want to fix the leaking-background issue I reported on them a while ago.

---------------------

You're right in that I don't support the the repeat(auto, ...) feature, and therefore Ex20 isn't working. I'll track this in a new open issue. That being said, while I agree on the Rachel's use case, I don't think its example will be working as she expect at all given the current spec. I already complained to www-style about it, I'll send a new mail to www-style with his example linked to show I'm not the only one disagreeing with the spec.

**Comment by mrego** [comment]
Thanks for reporting the issues, I agree with @FremyCompany that they're all in Chromium/Blink side.

**BACKGROUND PAINT ORDER**:

Regarding this issue, I've just created [a simple patch to fix it](https://codereview.chromium.org/978723002/) that should land soon. Thanks for the reminder!

**SHRINK-TO-FIT**:

It's true that current behavior is broken on Chromium, but not only for grids but also for flexboxes. I've reported [the bug](https://code.google.com/p/chromium/issues/detail?id=463847) and I hope it should be fixed soon.

**OUTLINE PAINT ORDER**:

Again @FremyCompany is right and once my patch to fix the **background paint order** lands, the outline in Chromium will have a weird behavior.

I've sent a [pull-request to Rachel](https://github.com/rachelandrew/grid-by-example/pull/4) with a possible solution to fix this (using `z-index`).

**REPEAT AUTO**:

This is not implemented yet in Chromium either.

And it's true that the example is wrong as [current spec](http://dev.w3.org/csswg/css-grid/#repeat-notation) doesn't allow you to use "auto" for a track breadth inside "repeat(auto)". You'll need to use fixed or percentage sizes there.

**Comment by FremyCompany** [comment]
Thanks for the confirmation, @mrego. 
I'm closing this issue as a separate issue has been opened for "repeat auto". I'll also open issues for the other remaining things to implement.

I hope @delebash is now convinced my polyfill is a good one ;-)

**Comment by delebash** [comment]
Thanks for the explanation and the quick fixes.
Yes I think your polyfill is Awsome, thanks very much for making this for everyone!!
